HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -201,76 +220,119 @@ void TritonClient::evaluate() {
-              //get results
-              std::shared_ptr<nic::InferResult> results_ptr(results);
-              bool status = triton_utils::warnIfError(results_ptr->RequestStatus(), "evaluate(): unable to get result");
-              if (!status) {
-                finish(false);
-                return;
-              }
-              auto t2 = std::chrono::high_resolution_clock::now();
-
-              if (!debugName_.empty())
-                edm::LogInfo(fullDebugName_)
-                    << "Remote time: " << std::chrono::duration_cast<std::chrono::microseconds>(t2 - t1).count();
-
-              const auto& end_status = getServerSideStatus();
-
-              if (verbose()) {
-                const auto& stats = summarizeServerStats(start_status, end_status);
-                reportServerSideStats(stats);
-              }
-
-              //check result
-              status = getResults(results_ptr);
-
-              //finish
-              finish(status);
-            },
-            options_,
-            inputsTriton_,
-            outputsTriton_),
-        "evaluate(): unable to launch async run");
-
-    //if AsyncRun failed, finish() wasn't called
-    if (!status)
-      finish(false);
+    try {
+      triton_utils::throwIfError(
+          client_->AsyncInfer(
+              [t1, start_status, this](nic::InferResult* results) {
+                //get results
+                std::shared_ptr<nic::InferResult> results_ptr(results);
+                try {
+                  triton_utils::throwIfError(results_ptr->RequestStatus(), "evaluate(): unable to get result");
+                } catch (...) {
+                  if (!checkEptr())
+                    return;
+                }
+                auto t2 = std::chrono::high_resolution_clock::now();
+
+                if (!debugName_.empty())
+                  edm::LogInfo(fullDebugName_)
+                      << "Remote time: " << std::chrono::duration_cast<std::chrono::microseconds>(t2 - t1).count();
+
+                if (verbose()) {
+                  inference::ModelStatistics end_status;
+                  try {
+                    end_status = getServerSideStatus();
+                  } catch (...) {
+                    if (!checkEptr())
+                      return;
+                  }
+
+                  const auto& stats = summarizeServerStats(start_status, end_status);
+                  reportServerSideStats(stats);
+                }
+
+                //check result
+                try {
+                  getResults(results_ptr);
+                } catch (...) {
+                  if (!checkEptr())
+                    return;
+                }
+
+                //finish
+                finish(true);
+              },
+              options_,
+              inputsTriton_,
+              outputsTriton_),
+          "evaluate(): unable to launch async run");
+    } catch (...) {
+      //if AsyncRun failed, finish() wasn't called

```

Comment:
Should this comment be
```suggestion
      //if AsyncInfer failed, finish() wasn't called

```
?